packages/shell-api/src/explainable-cursor.ts

```diff
@@ -15,10 +15,13 @@ export default class ExplainableCursor extends Cursor {
     this.verbosity = verbosity;
   }
 
-  async toReplString(): Promise<any> {
+  async asPrintable(): Promise<any> {
     return await this.cursor.explain(this.verbosity);
   }
-  shellApiType(): string {
-    return 'ExplainableCursor'; // required otherwise inherits Cursor's method
+  asShellResult(): ShellResult {
```

Comment:
Nice!

Reply:
I updated the code to use generics but we actually can't get around using <any> since we're adding the `asShellResult` method in the decorator, for all ShellApi classes so sometimes the generic will be string, sometimes object, sometimes other classes like `HelpProperties`.

It might make sense to revert if we're just going to be doing `ShellResult<any>` for the method.

Reply:
Also sometimes we return a promise of shell result and sometimes we return the result directly,  so I'm not sure the generics will actually work here